packages/opentelemetry-core/src/trace/NoopTracer.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Tracer, SpanOptions, Span, Propagator } from '@opentelemetry/types';
+import { NOOP_HTTP_TEXT_FORMAT } from '../context/propagation/NoopHttpTextFormat';
+import { NOOP_BINARY_FORMAT } from '../context/propagation/NoopBinaryFormat';
+import { NoopSpan } from './NoopSpan';
+
+export const NOOP_SPAN = new NoopSpan({ traceId: '', spanId: '' });
+
+/**
+ * No-op implementations of {@link Tracer}.
+ */
+export class NoopTracer implements Tracer {
```

Comment:
I understand that `NOOP_SPAN` is useful to return for non-sampled cases, but what would be the use case for `NoopTracer`?

If it's intended to just get some code started, what would you think about calling this TracerBase or similar? Would this be the class that eventually takes a scope storage in the constructor?

Reply:
The main reason for `NoopTracer`, at least for us, is when users disable the tracer then the implementation is switched to the noop to avoid unnecessary overhead. The noop tracer then uses only noop classes for the entire API surface.

Reply:
OK, makes sense!

Reply:
+1 on @rochdev comment (https://github.com/open-telemetry/opentelemetry-js/pull/98#discussion_r303030061)

And In OpenTracing there is a concept of [Global tracer](https://github.com/opentracing/opentracing-javascript/blob/master/src/global_tracer.ts#L44) (_which user can set using `initGlobalTracer()` function_). AFAIK Global tracer (`global.Tracer()`) should return minimal Tracer - `NoopTracer` if the app chooses to NOT use the SDK like basic-tracer or auto-tracer or vendor specific.